browser/brave_wallet/android/brave_wallet_native_worker.cc

```diff
@@ -111,25 +114,46 @@ void BraveWalletNativeWorker::ResetWallet(JNIEnv* env) {
 
 void BraveWalletNativeWorker::GetAssetPrice(
     JNIEnv* env,
-    const base::android::JavaParamRef<jstring>& asset) {
+    const base::android::JavaParamRef<jobjectArray>& from_assets,
+    const base::android::JavaParamRef<jobjectArray>& to_assets) {
+  std::vector<std::string> assets_from;
+  base::android::AppendJavaStringArrayToStringVector(env, from_assets,
+                                                     &assets_from);
+  std::vector<std::string> assets_to;
+  base::android::AppendJavaStringArrayToStringVector(env, to_assets,
+                                                     &assets_to);
+
   brave_wallet::BraveWalletService* brave_wallet_service =
       GetBraveWalletService();
   if (brave_wallet_service) {
     auto* asset_ratio_controller =
         brave_wallet_service->asset_ratio_controller();
     asset_ratio_controller->GetPrice(
-        base::android::ConvertJavaStringToUTF8(env, asset),
+        assets_from, assets_to,
         base::BindOnce(&BraveWalletNativeWorker::OnGetPrice,
                        weak_ptr_factory_.GetWeakPtr()));
   }
 }
 
-void BraveWalletNativeWorker::OnGetPrice(bool success,
-                                         const std::string& price) {
+void BraveWalletNativeWorker::OnGetPrice(
+    bool success,
+    std::vector<brave_wallet::mojom::AssetPricePtr> prices) {
+  std::string prices_json;
+  base::Value list(base::Value::Type::LIST);
+  for (const auto& asset_price : prices) {
```

Comment:
mojo types can be passed directly to java

Reply:
cc @deeppandya

Reply:
@deeppandya

Reply:
let's get wallet handler and java mojo work done and most likely the native worker will go away entirely

Reply:
I wanted to update it but since native worker will obsolete soon so I didn't update earlier. But I will update it now.